### PR TITLE
Update code_review_guidelines.md

### PR DESCRIPTION
**Pull request by rianfowler**
- updated to add code owners stuff 
- This should be merged after the code quality work is complete

### CODE REVIEW

File: platform/engineering/code_review_guidelines.md [hunk 1]
```diff
@@ -8,6 +8,21 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
...
+
+The code owners configuration can be found [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/CODEOWNERS). After creating your GitHub team, follow the conventions in the file to assign your team to an application folder. You should understand the following when making changes to the code owners configuration: 
+
+- The **last matching rule** that matches all files in a change **takes precedence**.
+- Application code should be structured to **require as few rules as possible**. 
+- Only one required reviewer at a time can be triggered by code owners rules.
+- Code owners rules are applied to every push after a PR is created. Required reviewers can be removed during this process but **GitHub does not remove the front end review group once it has been added to a PR** as a required reviewer.
```
Comment (cvalarida) [inline platform/engineering/code_review_guidelines.md]:
```suggestion
- Code owners rules are applied to every push after a PR is created. Required reviewers can be removed during this process but **GitHub does not remove the frontend review group once it has been added to a PR** as a required reviewer. 
```
We probably should decide `frontend` vs `front end`, but you've been using `frontend` the rest of the time in this section. :man_shrugging:

File: platform/engineering/code_review_guidelines.md [hunk 2]
```diff
@@ -8,6 +8,21 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
...
+
+The code owners configuration can be found [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/CODEOWNERS). After creating your GitHub team, follow the conventions in the file to assign your team to an application folder. You should understand the following when making changes to the code owners configuration: 
+
+- The **last matching rule** that matches all files in a change **takes precedence**.
+- Application code should be structured to **require as few rules as possible**. 
+- Only one required reviewer at a time can be triggered by code owners rules.
+- Code owners rules are applied to every push after a PR is created. Required reviewers can be removed during this process but **GitHub does not remove the front end review group once it has been added to a PR** as a required reviewer. 
+- The frontend review group approves changes to code owners file. GitHub teams will be added to the vets-website repository during this review- this is required for code owners rules to trigger.
```
Comment (cvalarida) [inline platform/engineering/code_review_guidelines.md]:
```suggestion
- The frontend review group approves changes to code owners file. GitHub teams will be added to the vets-website repository during this review—this is required for code owners rules to trigger. 
```

**Review by cvalarida — APPROVED** [review]
Looks great! I have no substantive suggestions, just a couple o' copy edit comments.

I dig it.

File: platform/engineering/code_review_guidelines.md [hunk 3]
```diff
@@ -8,6 +8,21 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
...
 
+## Code owners on vets-website
+
+VSP uses GitHub's [code owners](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) feature to trigger required reviews. A code owner is a GitHub [organization](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/about-organizations) that must approve a change before it can be merged. 
+
+By default, the [frontend review group](https://github.com/orgs/department-of-veterans-affairs/teams/frontend-review-group) is the code owner of vets-website. VSA teams can optionally create a GitHub team under the [Department of Veteran Affairs GitHub organization](https://github.com/orgs/department-of-veterans-affairs/teams) and assign that team as a code owner of application [folders](https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/applications). **This will remove the frontend review group as a required reviewer when changes are made to those folders.**
```
Comment (gunsch) [inline platform/engineering/code_review_guidelines.md]: optional stylistic thing: backticks for proper technical names? `frontend-review-group` and `vets-website` (definitely the latter, less sure on the former)

**Comment by johnpaulashenfelter** [comment]
@rianfowler seems like we need the parallel edits for backend? Not that that need would block this.

I tend to be in agreement with @gunsch that maybe there's an overview that belongs here about each repo having a CODEOWNERS and then moving the details into a table(?) or other document. While `vets-website` and `vets-api` are both the most commonly reviewed repos, there are edits to other repos that probably need a callout. 

I'm less concerned about the size of individual docs -- but i do think there needs to be some mention of each repo we end up reviewing, even if its rare. The reviews of endpoints to the fwdproxy spring to mind since they show up in `devops` which has no real idea about whether they're sensible or not. And it might be that part of the output from this is a TBD for each of the repos each team covers.

**Comment by rianfowler** [comment]
@johnpaulashenfelter I think we can follow-up and add a table later. The goal of this is to get VFS teams to own their own code. It probably doesn't matter to them how we enforce code reviews on other repos- only that it's the default behavior for the platform owned code.

**Pull request merged**